### PR TITLE
Update continuous integration practice

### PR DESCRIPTION
**Pull request by stefaniuk**
Changes:
- Update continuous integration practice document
- Add ToC to all docs in the practice directory

### CODE REVIEW

File: practices/continuous-integration.md [hunk 1]
```diff
@@ -1,52 +1,132 @@
...
+| Full build                    | Do you have a full/overnight build that runs everything? How do you manage any failures from that?                                                                                                                                                                                                                                                                 |
+| Failure rate                  | Are your builds reliable, or do you get a high rate of false failures?                                                                                                                                                                                                                                                                                             |
+| Result publishing             | Are the results from your build (success/error/warn) clear and concise?  Can you always tell exactly what caused a build to fail? Do you get spammed with unnecessary emails? Do you use slack/etc. for build alerting? Are there manual steps that could be automated? e.g. telling Test that a build is ready, switching a route53, clearing up old deployments. |
```
Comment (andyblundell) [inline practices/continuous-integration.md:91]: Can we rephrase "telling Test that a build is ready" pls - sounds a bit hand-off-y :)

File: practices/continuous-integration.md [hunk 2]
```diff
@@ -1,52 +1,132 @@
...
+| Effectiveness and Maintenance | Despite your fancy build pipeline, are you still seeing high bug/error/security/etc. rates in production? Are you spending too much team effort on maintaining your pipelines? Are they too complex without delivering benefits?                                                                                                                                   |
```
Comment (andyblundell) [inline practices/continuous-integration.md:92]: Can we reword "fancy" pls?

File: practices/continuous-integration.md [hunk 3]
```diff
@@ -1,52 +1,132 @@
...
+| Stress tests             | Have you tested to see where your application's bottleneck is? How often do you repeat these tests? Are they fully automated? Do you/Texas have the tools to automate this?                                                                                                                                                                                                                  |
+| Load tests               | Do you frequently check that your application can cope with as-live load? Do you know how much increased load you can handle? Are these tests fully automated as part of your pipeline? How often do you repeat these tests? Do you/Texas have the tools to automate this?                                                                                                                   |
```
Comment (andyblundell) [inline practices/continuous-integration.md:102]: Let's drop the Texan reference :)

File: practices/continuous-integration.md [hunk 4]
```diff
@@ -1,52 +1,132 @@
...
 
-* **Deploy what you tested**
-  Never rebuild a deployment artefact for final release.
+### Pipeline steps
 
-* **Automate quality gates**
-  Quick-running, reliable automate tests allow you to refactor. Use that reduction in risk to regularly refactor and improve code.
-  Quality gates (such as [tests](testing.md)) should be applied and evidenced by the build pipeline, not via manual processes.
+| Pipeline steps           | Things to consider                                                                                                                                                                                                                                                                                                                                                                           |
```
Comment (andyblundell) [inline practices/continuous-integration.md:96]: Lots of overlap with the quality-checks page, probably worth adding a link across (or trying to combine the lists)?

File: practices/continuous-integration.md [hunk 5]
```diff
@@ -1,52 +1,132 @@
...
+| FHIR compliance          | Do you check your FHIR APIs for FHIR compliance? Are these fully automated as part of your pipeline? What automated tools do you use? Does the tooling integrate directly into your IDE for instant feedback?                                                                                                                                                                                |
+| Tech RAG checking        | Have your frameworks and technologies been deprecated by NHSD? Do you currently check your application against the NHSD Tech RAG? If so, how? Does the tooling integrate directly into your IDE for instant feedback?                                                                                                                                                                        |
```
Comment (andyblundell) [inline practices/continuous-integration.md:110]: Tech RAG == Tech radar?

**Review by andyblundell — CHANGES_REQUESTED** [review]
All looks good - a few suggested tweaks to the list you've incorporated

File: practices/continuous-integration.md (hunk 1 above)
Comment (stefaniuk) [reply practices/continuous-integration.md:91]: I removed this sentence.

File: practices/continuous-integration.md (hunk 2 above)
Comment (stefaniuk) [reply practices/continuous-integration.md:92]: Reworded to "Despite having a robust build pipeline"

File: practices/continuous-integration.md (hunk 3 above)
Comment (stefaniuk) [reply practices/continuous-integration.md:102]: The Texan reference dropped

File: practices/continuous-integration.md (hunk 5 above)
Comment (stefaniuk) [reply practices/continuous-integration.md:110]: Done

File: practices/continuous-integration.md (hunk 4 above)
Comment (stefaniuk) [reply practices/continuous-integration.md:96]: There is more work planned in this area ("continuous integration") as we are reviewing our practice with one or two of the programmes. The intention is to incorporate any given feedback here. This should follow as a separate PR. I'll bear that in mind to link to the quality checks page.

**Pull request merged**